### PR TITLE
build: Use Loader Script instead of `@sentry/gatsby`

### PR DESCRIPTION
**Pull request by mydea**
This replaces the usage of `@sentry/gatbsy` with usage of the Loader Script.

This is done so we use the Loader Script in an actual application, allowing us to verify this. The dedicated gatsby package does not add a lot on top of what the loader offers, and we still dogfood it in the develop docs.

Note that I noticed that we have already been providing a custom webpack plugin config here: https://github.com/getsentry/sentry-docs/blob/master/src/gatsby/onCreateWebpackConfig.ts#L15

So this should still work the same, actually.

I removed the manually set `release`, letting this being set by the webpack plugin (hopefully). The DSN is still read from `GATSBY_SENTRY_DSN`, the same as before.

We still depend on `@sentry/browser`, but only for the types.

Closes https://github.com/getsentry/sentry-docs/issues/6884

### CODE REVIEW

**Comment by mydea** [comment]
Hmm, is there a way to test this with Sentry enabled in a vercel preview? 🤔

File: package.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
     "@mdx-js/mdx": "^1.6.18",
     "@mdx-js/react": "^1.6.18",
     "@sentry-internal/global-search": "^0.5.7",
-    "@sentry/gatsby": "^7.46.0",
+    "@sentry/browser": "7.51.0",
```
[the remark this answers is not in the document]
Comment (mydea) [reply package.json:20]: just for types, as we access `Sentry.getCurrentHub()` in few places. Not sure, we can also get rid of this and just tread these as `any` 🤔 WDYT?

[the remark this answers is not in the document]
Comment (mydea) [reply package.json:20]: jup, true! Maybe if we want to push the loader further, we can think about having a `@sentry/loader` package that provides some shims on top of `window` + types - also, as in this PR, if you use the loader and want to access some APIs in typed/module code, it's a bit of a pain as you need to do a lot of guarding etc. manually.

[the remark this answers is not in the document]
Comment (bruno-garcia) [reply package.json:20]: I think the loader package is a good idea, because this will be confusing (folks might open PR bumping via yarn)

**Review by HazAT — APPROVED** [review]

**Review by shanamatthews — APPROVED** [review]

**Pull request merged**